src/vm/moar/QAST/QASTCompilerMAST.nqp

```diff
@@ -303,6 +303,42 @@ my class MASTCompilerInstance {
                 $out++;
                 $block := $block.outer;
             }
+            my $buf8 := create_buf(uint8);
+            my sub convert-two($str) {
+                my $str2 := nqp::encode($str, 'iso-8859-1', $buf8.new);
+                return nqp::decode($str2, 'utf8');
+            }
+            my sub convert($str) {
+                my $str2 := nqp::encode($str, 'iso-8859-1', $buf8.new);
+                return nqp::decode($str2, 'utf8');
+            }
+            my sub create_buf($type) {
+                my $buf := nqp::newtype(nqp::null(), 'VMArray');
+                nqp::composetype($buf, nqp::hash('array', nqp::hash('type', $type)));
+                nqp::setmethcache($buf, nqp::hash('new', method () {nqp::create($buf)}));
+                $buf;
+            }
+            my $handle-mangled-strings := 0;
```

Comment:
Is this a debugging thing?  I don't understand the `if` just after it?

Reply:
Yes it is. It's also so it's easy to see how this proof of concept was achieved when viewing the pull request.